src/libdredd/src/mutate_visitor.cc

```diff
@@ -99,17 +103,17 @@ bool MutateVisitor::VisitBinaryOperator(
   // In particular, we do not want to mess with pointer arithmetic.
   const auto* result_type =
       binary_operator->getType()->getAs<clang::BuiltinType>();
-  if (result_type == nullptr || !result_type->isInteger()) {
+  if (IsTypeSupported(result_type)) {
```

Comment:
Given that `result_type`, `lhs_type` and `rhs_type` are never used again, it would be better to change `IsTypeSupported` so that it accepts a `QualType` as an argument, and then you can just do:

`if (IsTypeSupported(binary_operator->getType())`

and similar below with `getType()` on the LHS and RHS.

The `getAs<clang::BuiltinType>()` can then go inside the `IsTypeSupported` method.